scripts/dev

```diff
@@ -0,0 +1,202 @@
+#! /usr/bin/ruby
+
+require "rake"
+require "irb"        # debugging
+require "io/console" # for IO#getch
+
+# ExtraArguments is a hack that alters how the command line arguments are parsed.
+#
+# Rake has a few ways to handle arguments, but the simplest one uses bracket characters
+# which are problematic in the default zsh config and the others require using flags.
+#
+# With this code, a task can specify :consume_args as a prerequisite and any options after
+# the task name will be pulled into that task's args. You can only specify a single task
+# to run in this case, but that's the most common usage anyhow.
+module ExtraArguments
+  def execute(args)
+    if prereqs.include? "consume_args"
+      new_args = Rake::TaskArguments.new(args.names, args.to_a.concat($extra_args || []))
+      super new_args
+    else
+      super
+    end
+  end
+end
+
+class Rake::Task
+  prepend ExtraArguments
+end
+
+@needed = []
+def needs(executable, install_with:nil)
+  @needed << [executable, install_with]
+end
+
+needs "air", install_with: "go get -u github.com/cosmtrek/air"
+needs "yarn"
+needs "psql", install_with: "brew install postgresql"
+needs "docker-compose"
+needs "go"
+needs "gotest", install_with: "go get -u github.com/rakyll/gotest"
+needs "goimports", install_with: "go get -u golang.org/x/tools/cmd/goimports"
+needs "ulimit"
+
+Rake.application.init("scripts/dev")
+
+def ask_to_install(executable, install_with)
+  puts "Do you want to install it? (Y/n)"
+  response = STDIN.getch
+  if ["y", "Y", "\r"].include?(response)
+    sh install_with
+    check_bin(executable)
+    # TODO let the user know that their path is not configured correctly
+  end
+end
+
+def check_bin(executable, install_with: nil)
+  if `which #{executable}` == ""
+    puts "#{executable} not found on path"
+    if install_with
+      ask_to_install(executable, install_with)
+    end
+  end
+end
+
+desc "Check to see if the app's prerequisites are installed"
+task :prereqs do
+  @needed.each do |executable, install_with|
+    check_bin(executable, install_with: install_with)
+  end
+end
+
+task :consume_args do |t, args|
+  $extra_args = Rake.application.top_level_tasks.slice!(1..-1)
+end
+
+
+### Tasks
+
+desc "Resets application to an empty state"
+task :reset do
+  # close any hanging connections to the database
+  `echo "select pg_terminate_backend(pg_stat_activity.pid) from pg_stat_activity where datname='postgres' AND state='idle'" | psql`
+
+  # drop and recreate the database
+  sh "dropdb postgres && createdb postgres"
```

Comment:
We might be able to drop and recreate the database via docker-compose:

```
docker-compose kill db && docker-compose rm -fv db && docker-compose up -d db
```

That could be another rake task like `migrate`.